src/conf.py

```diff
@@ -1215,33 +1215,87 @@ def makeBanmask(self, hostmask, options=None, channel=None, network=None):
         isn't specified via options, the value of
         conf.supybot.protocols.irc.banmask is used.
 
+        Unlike :meth:`makeExtBanmasks`, this is guaranteed to return an
+        RFC1459-like mask, suitable for ircdb's ignore lists.
+
         options - A list specifying which parts of the hostmask should
         explicitly be matched: nick, user, host.  If 'exact' is given, then
-        only the exact hostmask will be used."""
-        if not channel:
-            channel = dynamic.channel
+        only the exact hostmask will be used.
+        """
         if not network:
             network = dynamic.irc.network
+        if not options:
+            options = supybot.protocols.irc.banmask.getSpecific(
+                network, channel)()
+        options = [option for option in options if option != 'account']
+        print(hostmask, options)
+        masks = self.makeExtBanmasks(
+            hostmask, options, channel, network=network)
+        assert len(masks) == 1, 'Unexpected number of banmasks: %r' % masks
+        print(masks)
+        return masks[0]
+
+    def makeExtBanmasks(self, hostmask, options=None, channel=None, *, network):
```

Comment:
`s/options/banstyles/` perhaps?

Reply:
it's meant to be existent with the existing `makeBanmask` :/

Reply:
Ah I missed that, not a big deal then